routers/web/repo/issue.go

```
@@ -1042,7 +1042,8 @@ func NewIssuePost(ctx *context.Context) {
 		return
 	}
 
-	if projectID > 0 {
+	// User must also be able to see the project.
+	if projectID > 0 && ctx.Repo.CanRead(unit.TypeProjects) {
```

```
if projectID > 0 && !ctx.Repo.CanRead(unit.TypeProjects)
```
We need to return an error.